raster/r.terraflow/unionFind.h

```diff
@@ -122,23 +129,35 @@ template <class T>
 inline void unionFind<T>::makeSet(T x)
 {
     /* cout << "makeSet " << x << "\n"; print(); */
-    assert(x > 0);
+    if (x <= 0) {
+        G_fatal_error(_("Invalid set element: %d"), x);
+    }
     if (x >= maxsize) {
         /* reallocate parent */
         cout << "UnionFind::makeSet: reallocate double " << maxsize << "\n";
-        parent = (T *)realloc(parent, 2 * maxsize * sizeof(T));
-        assert(parent);
-        memset(parent + maxsize, 0, maxsize * sizeof(T));
-        /*reallocate rank */
-        rank = (T *)realloc(rank, 2 * maxsize * sizeof(T));
-        assert(rank);
-        memset(rank + maxsize, 0, maxsize * sizeof(T));
+        if (void *new_parent = std::realloc(parent, 2 * maxsize * sizeof(T))) {
+            parent = static_cast<T *>(new_parent);
+            std::memset(parent + maxsize, 0, maxsize * sizeof(T));
+        }
+        else {
+            G_fatal_error(_("Not enough memory for %s"), "parent");
+        }
+        /* reallocate rank */
+        if (void *new_rank = std::realloc(rank, 2 * maxsize * sizeof(T))) {
+            rank = static_cast<T *>(new_rank);
+            std::memset(rank + maxsize, 0, maxsize * sizeof(T));
+        }
+        else {
+            G_fatal_error(_("Not enough memory for %s"), "rank");
+        }
         /*update maxsize */
         maxsize *= 2;
     }
     /*since x is disjoint we require x not be already in the set; should
       relax this..*/
-    assert(!inSet(x));
+    if (inSet(x)) {
```

Comment:
Failure of `calloc()` and `realloc()` are runtime failures, which are fatal. But the other uses of assert() in this file are debugging checks (asserts). `assert()` does nothing if `NDEBUG` is defined. We should keep them as they were. Please revert the these changes (not related to calloc or realloc), but change `#include <assert.h>`  to `#include <cassert>`.

Reply:
done.
Thanks for the info